modules/debug_cmd/debug_cmd.c

```diff
@@ -160,6 +160,143 @@ static int cmd_play_file(struct re_printf *pf, void *arg)
 }
 
 
+struct ausrc_st;
+
+
+struct fileinfo_st {
+	struct ausrc_st *ausrc;
+	struct ausrc_prm prm;
+	size_t sampc;
+	struct tmr tmr;
+	bool   finished;
+};
+
+
+static void fileinfo_destruct(void *arg)
+{
+	struct fileinfo_st *st = arg;
+
+	tmr_cancel(&st->tmr);
+	mem_deref(st->ausrc);
+}
+
+
+static void fileinfo_timeout(void *arg)
+{
+	struct fileinfo_st *st = arg;
+	size_t ms = 0;
+
+	if (st->prm.ch && st->prm.srate)
+		ms = st->sampc * 1000 / st->prm.ch / st->prm.srate;
+
+	if (st->finished) {
+		info("debug_cmd: length = %u ms\n", ms);
```

Comment:
I think that the unit _seconds_ is more user-friendly, use e.g. `double seconds`.

length: 30500 ms
length: 30.5 seconds